.zsh/prompt.zsh

```diff
@@ -83,23 +70,26 @@ ZSH_HIGHLIGHT_STYLES[comment]='fg=white,bold'
   typeset -g POWERLEVEL9K_SHORTEN_FOLDER_MARKER="(${(j:|:)anchor_files})"
   typeset -g POWERLEVEL9K_SHORTEN_DIR_LENGTH=1
   typeset -g POWERLEVEL9K_DIR_MAX_LENGTH=80
-  typeset -g POWERLEVEL9K_DIR_MIN_COMMAND_COLUMNS=40
-  typeset -g POWERLEVEL9K_DIR_MIN_COMMAND_COLUMNS_PCT=50
-
-  typeset -g POWERLEVEL9K_DIR_SHOW_WRITABLE=true
-  typeset -g POWERLEVEL9K_DIR_CLASSES=()
 
   #####################################[ vcs: git status ]######################################
-  typeset -g POWERLEVEL9K_VCS_BRANCH_ICON='\uF126 '
-
+  function _git_dirty() {
+    (( VCS_STATUS_STASHES        ||
+       VCS_STATUS_NUM_CONFLICTED ||
+       VCS_STATUS_NUM_UNSTAGED   ||
+       VCS_STATUS_NUM_UNTRACKED  ||
+       VCS_STATUS_NUM_STAGED     ||
+       VCS_STATUS_COMMITS_BEHIND ||
+       VCS_STATUS_COMMITS_AHEAD ))
+  }
+  functions -M _git_dirty 0 0
```

Comment:
This is great, very nice and readable, I'm considering to maybe go away from expansions entirely and move everything in this function, so that outside remains something like `vcs=$((_git_status))` - would you say it's okay to do performance-wise?

Also, for some reason this right now doesn't work (it seems to always return the same value, so space is always added), but it won't be an issue anyway if I move everything in the function.